The ffmpeg plugin uses functions from libavcodec, so ensure that it is linked
against libavcodec.

Commit 9bde48f1b59d upstream.

Also make sure that the ffmpeg.c build test uses -std=gnu99, as does the
final build.

Index: configure
--- configure.orig
+++ configure
@@ -478,7 +478,7 @@ check_ffmpeg()
 	check_header "libswresample/swresample.h" $FFMPEG_CFLAGS || return $?
 	# ffmpeg api changes so frequently that it is best to compile the module
 	libs="$LDDLFLAGS $FFMPEG_LIBS"
-	cflags="$SOFLAGS $FFMPEG_CFLAGS"
+	cflags="-std=gnu99 $SOFLAGS $FFMPEG_CFLAGS"
 	topdir=`dirname "$0"`
 	ffmpeg_code=`cat "$topdir"/ip/ffmpeg.c | sed 's/\\\n//g'`
 	msg_checking "for successful build of ffmpeg.c"
